frappe/__init__.py

```diff
@@ -1644,10 +1644,19 @@ def setup_module_map():
 
 	if not (local.app_modules and local.module_app):
 		local.module_app, local.app_modules = {}, {}
-		for app in get_all_apps(with_internal_apps=True):
+
+		# If a site is initialized, only get the apps installed on it
+		if getattr(local, "site"):
+			apps = get_installed_apps(_ensure_on_bench=True)
+		else:
+			apps = get_all_apps(with_internal_apps=True)
```

Comment:
this block would never get executed.. `frappe.local.site` is always set in all places where this is actually used. let's just make "local.site" a requirement for this function instead.

Reply:
Calling any `bench` command without specifying a site (relying on a [default site](https://github.com/frappe/frappe/blob/develop/frappe/utils/bench_helper.py#L56)) leads to this since it has to call init to read the common site configuration.

(Maybe could just skip setting up the module map in init when a site is not passed?)

Reply:
Much cleaner now.